lib/Importer/ONNX.cpp

```diff
@@ -502,6 +502,37 @@ std::unique_ptr<ONNXModelLoader> ONNXModelLoader::parse(const void *onnxModel,
 ONNXModelLoader::ONNXModelLoader(Function &F)
     : CommonOperatorLoader({}, {}, F) {}
 
+void ONNXModelLoader::CheckIO(onnx::GraphProto &net,
+                              llvm::ArrayRef<const char *> names,
+                              llvm::ArrayRef<Tensor *> tensors)
+{
+  for (unsigned i = 0; i < names.size(); i++) {
+    // Look if a corresponding input exists
+    for(int j = 0; j < net.input_size(); j++) {
+      const onnx::ValueInfoProto& value_info = net.input(j);
+      const std::string& input_name = value_info.name();
+
+      if (input_name == names[i]) {
+        llvm::ArrayRef<size_t> dims = tensors[i]->dims();
+        const onnx::TensorShapeProto& shape = value_info.type().tensor_type().shape();
+
+        // Check
+        if (dims.size() != shape.dim_size()) {
+          llvm::outs().flush();
+          llvm_unreachable("Mismatch between input image and ONNX input shape");
+        }
+        // Allow batch dimensions to be different.
+        for(unsigned k = 1; k < dims.size(); k++) {
+          if (dims[k] != shape.dim(k).dim_value()) {
+            llvm::outs().flush();
```

Comment:
Why do we need the flush?

Reply:
Removed. It was not supposed to be here.